### ISSUE TITLE
ODBC driver indicated must indicate string length correctly.: FromBytesWithNulError { kind: InteriorNul(38) }

### ISSUE THREAD

**Issue by AdrianSkierniewski** [pacman82/odbc2parquet#36]
Hi,

First of all, thank you for such a nice tool!
I'm trying to use it to dump the whole database from the MS SQL 15 Server, but unfortunately, I'm getting some error on the biggest table there. I tried to split it into smaller chunks, use different drivers but nothing actually worked.

```
thread 'main' panicked at 'ODBC driver indicated must indicate string length correctly.: FromBytesWithNulError { kind: InteriorNul(38) }', C:\Users\runneradmin\.cargo\registry\src\github.com-1ecc6299db9ec823\odbc-api-0.14.0\src\buffers\text_column.rs:56:22
```

I tested it on a couple of other tables and everything worked. Do you have any idea what could cause that?

**Comment by pacman82** [comment]
Hello, thanks for having nice words about my tool, despite experiencing what I feel is a bug.

The panic in question can occurr in the upstream `odbc-api` here: <https://github.com/pacman82/odbc-api/blob/1f0a3fc648ce71a00f4541b46a5585895f16a16e/odbc-api/src/buffers/text_column.rs#L55>

The ODBC driver is supposed put the length of the string in the indicator buffer. There should not be a terminating zero before that. However Microsoft's ODBC driver so far gave me very little trouble, so I am hesitant to assume the errors lies with them. I could see the above piece of code failing in a situation, there a string is truncated and the driver is trying to do some clever thing, not to truncate a multibyte UTF-8 symbol. However this tool should always allocate enough memory so truncation never happens at all. In short I have a hunch, but to reproduce it I would really like some extra info.

1. If you could kindly run the tool with more very verbose logging using the `-vvvv` flag before the `query` command and post the output. I am especially interssted in schema information and buffer sizes.
2. Do you happen to know what encoding your system locale uses? (On Linux and OS-X you may assume it to be UTF-8).
3. Is the error caused by a specific value. (Maybe try adding a `LIMIT 1` to the SQL query. If the error appears on the first row it is likely to appear with any input).
4. It would be helpful to know which column causes the issue.

I am fully aware that generating these informations may take time. Please read this just as a list of inputs which might be helpful to me. Even a subset of these information may be extremly helpful (especially the verbose output). So please do not spend too much time on generating a certain input for me.

**Comment by AdrianSkierniewski** [comment]
Hi @pacman82 

Thank you for such a quick response to this issue. I'll try to answer all your questions below:
1. Output with verbose flag
```bash
2021-02-11T08:13:35+00:00 - DEBUG - ODBC Environment created.
2021-02-11T08:13:35+00:00 - WARN - State: 01000, Native error: 0, Message: [Microsoft][ODBC Driver Manager] The driver doesn't support the version of ODBC behavior that the application requested (see SQLSetEnvAttr).
2021-02-11T08:13:35+00:00 - WARN - State: 01000, Native error: 5701, Message: [Microsoft][ODBC SQL Server Driver][SQL Server]Changed database context to 'DEMO_DB'.
2021-02-11T08:13:35+00:00 - WARN - State: 01000, Native error: 5703, Message: [Microsoft][ODBC SQL Server Driver][SQL Server]Changed language setting to us_english.
2021-02-11T08:13:35+00:00 - INFO - Batch size set to 100000
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 1: ColumnDescription { name: [68, 73, 77, 95, 79, 82, 68, 69, 82, 95, 80, 79, 83, 73, 84, 73, 79, 78], data_type: WVarchar { length: 133 }, nullability: NoNulls }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 1: BufferDescription { nullable: true, kind: Text { max_str_len: 532 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 2: ColumnDescription { name: [75, 69, 89, 95, 79, 82, 68, 69, 82, 95, 80, 79, 83, 73, 84, 73, 79, 78], data_type: WVarchar { length: 100 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 2: BufferDescription { nullable: true, kind: Text { max_str_len: 400 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 3: ColumnDescription { name: [84, 88, 84, 95, 71, 82, 79, 85, 80, 73, 78, 71, 95, 65, 95, 49], data_type: WVarchar { length: 100 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 3: BufferDescription { nullable: true, kind: Text { max_str_len: 400 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 4: ColumnDescription { name: [84, 88, 84, 95, 71, 82, 79, 85, 80, 73, 78, 71, 95, 65, 95, 50], data_type: WVarchar { length: 100 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 4: BufferDescription { nullable: true, kind: Text { max_str_len: 400 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 5: ColumnDescription { name: [84, 88, 84, 95, 71, 82, 79, 85, 80, 73, 78, 71, 95, 65, 95, 51], data_type: WVarchar { length: 100 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 5: BufferDescription { nullable: true, kind: Text { max_str_len: 400 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 6: ColumnDescription { name: [75, 69, 89, 95, 79, 82, 68, 69, 82], data_type: WVarchar { length: 100 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 6: BufferDescription { nullable: true, kind: Text { max_str_len: 400 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 7: ColumnDescription { name: [84, 88, 84, 95, 79, 82, 68, 69, 82, 95, 80, 79, 83, 73, 84, 73, 79, 78], data_type: WVarchar { length: 500 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 7: BufferDescription { nullable: true, kind: Text { max_str_len: 2000 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 8: ColumnDescription { name: [84, 88, 84, 95, 79, 82, 68, 69, 82, 95, 80, 79, 83, 73, 84, 73, 79, 78, 95, 76, 79, 78, 71], data_type: WVarchar { length: 500 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 8: BufferDescription { nullable: true, kind: Text { max_str_len: 2000 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 9: ColumnDescription { name: [68, 73, 77, 95, 68, 65, 84, 69], data_type: WVarchar { length: 100 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 9: BufferDescription { nullable: true, kind: Text { max_str_len: 400 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 10: ColumnDescription { name: [68, 73, 77, 95, 73, 78, 67, 79, 84, 69, 82, 77], data_type: WVarchar { length: 100 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 10: BufferDescription { nullable: true, kind: Text { max_str_len: 400 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 11: ColumnDescription { name: [84, 88, 84, 95, 73, 78, 67, 79, 84, 69, 82, 77, 95, 65, 68, 68, 73, 84, 73, 79, 78], data_type: WVarchar { length: 500 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 11: BufferDescription { nullable: true, kind: Text { max_str_len: 2000 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 12: ColumnDescription { name: [68, 73, 77, 95, 68, 79, 67, 85, 77, 69, 78, 84, 95, 84, 89, 80, 69], data_type: WVarchar { length: 242 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 12: BufferDescription { nullable: true, kind: Text { max_str_len: 968 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 13: ColumnDescription { name: [84, 88, 84, 95, 83, 84, 65, 84, 85, 83, 95, 79, 82, 68, 69, 82], data_type: WVarchar { length: 500 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 13: BufferDescription { nullable: true, kind: Text { max_str_len: 2000 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 14: ColumnDescription { name: [84, 88, 84, 95, 83, 84, 65, 84, 85, 83, 95, 79, 82, 68, 69, 82, 95, 80, 79, 83, 73, 84, 73, 79, 78], data_type: WVarchar { length: 500 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 14: BufferDescription { nullable: true, kind: Text { max_str_len: 2000 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 15: ColumnDescription { name: [84, 88, 84, 95, 65, 83, 83, 73, 71, 78, 77, 69, 78, 84, 95, 67, 65, 84, 69, 71, 79, 82, 89], data_type: WVarchar { length: 500 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 15: BufferDescription { nullable: true, kind: Text { max_str_len: 2000 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 16: ColumnDescription { name: [75, 69, 89, 95, 85, 79, 77], data_type: WVarchar { length: 100 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 16: BufferDescription { nullable: true, kind: Text { max_str_len: 400 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 17: ColumnDescription { name: [75, 69, 89, 95, 68, 69, 76, 73, 86, 69, 82, 89, 95, 68, 65, 84, 69], data_type: WVarchar { length: 100 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 17: BufferDescription { nullable: true, kind: Text { max_str_len: 400 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 18: ColumnDescription { name: [78, 85, 77, 95, 81, 85, 65, 78, 84, 73, 84, 89], data_type: WVarchar { length: 20 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 18: BufferDescription { nullable: true, kind: Text { max_str_len: 80 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 19: ColumnDescription { name: [78, 85, 77, 95, 81, 85, 65, 78, 84, 73, 84, 89, 95, 70, 65, 67, 84, 79, 82], data_type: WVarchar { length: 20 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 19: BufferDescription { nullable: true, kind: Text { max_str_len: 80 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 20: ColumnDescription { name: [75, 69, 89, 95, 79, 82, 68, 69, 82, 95, 80, 79, 83, 73, 84, 73, 79, 78, 95, 85, 73], data_type: WVarchar { length: 100 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 20: BufferDescription { nullable: true, kind: Text { max_str_len: 400 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 21: ColumnDescription { name: [68, 73, 77, 95, 83, 89, 83, 84, 69, 77], data_type: WVarchar { length: 100 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 21: BufferDescription { nullable: true, kind: Text { max_str_len: 400 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 22: ColumnDescription { name: [84, 88, 84, 95, 77, 65, 84, 69, 82, 73, 65, 76], data_type: WVarchar { length: 500 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 22: BufferDescription { nullable: true, kind: Text { max_str_len: 2000 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 23: ColumnDescription { name: [84, 88, 84, 95, 77, 65, 84, 69, 82, 73, 65, 76, 95, 76, 79, 78, 71], data_type: WVarchar { length: 500 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 23: BufferDescription { nullable: true, kind: Text { max_str_len: 2000 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 24: ColumnDescription { name: [84, 88, 84, 95, 67, 82, 69, 65, 84, 79, 82], data_type: WVarchar { length: 500 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 24: BufferDescription { nullable: true, kind: Text { max_str_len: 2000 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 25: ColumnDescription { name: [68, 73, 77, 95, 77, 65, 84, 69, 82, 73, 65, 76], data_type: WVarchar { length: 133 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 25: BufferDescription { nullable: true, kind: Text { max_str_len: 532 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 26: ColumnDescription { name: [84, 88, 84, 95, 77, 65, 84, 69, 82, 73, 65, 76, 95, 71, 82, 79, 85, 80], data_type: WVarchar { length: 500 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 26: BufferDescription { nullable: true, kind: Text { max_str_len: 2000 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 27: ColumnDescription { name: [84, 88, 84, 95, 77, 65, 84, 69, 82, 73, 65, 76, 95, 71, 82, 79, 85, 80, 95, 76, 79, 78, 71], data_type: WVarchar { length: 500 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 27: BufferDescription { nullable: true, kind: Text { max_str_len: 2000 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 28: ColumnDescription { name: [68, 73, 77, 95, 77, 65, 84, 69, 82, 73, 65, 76, 95, 71, 82, 79, 85, 80], data_type: WVarchar { length: 133 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 28: BufferDescription { nullable: true, kind: Text { max_str_len: 532 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 29: ColumnDescription { name: [84, 88, 84, 95, 83, 85, 80, 80, 76, 73, 69, 82], data_type: WVarchar { length: 500 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 29: BufferDescription { nullable: true, kind: Text { max_str_len: 2000 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 30: ColumnDescription { name: [84, 88, 84, 95, 83, 85, 80, 80, 76, 73, 69, 82, 95, 76, 79, 78, 71], data_type: WVarchar { length: 500 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 30: BufferDescription { nullable: true, kind: Text { max_str_len: 2000 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 31: ColumnDescription { name: [68, 73, 77, 95, 83, 85, 80, 80, 76, 73, 69, 82], data_type: WVarchar { length: 133 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 31: BufferDescription { nullable: true, kind: Text { max_str_len: 532 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 32: ColumnDescription { name: [68, 73, 77, 95, 67, 79, 78, 84, 82, 65, 67, 84], data_type: WVarchar { length: 133 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 32: BufferDescription { nullable: true, kind: Text { max_str_len: 532 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 33: ColumnDescription { name: [70, 76, 71, 95, 70, 73, 78, 65, 76, 95, 68, 69, 76, 73, 86, 69, 82, 89], data_type: Bit, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 33: BufferDescription { nullable: true, kind: Bit }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 34: ColumnDescription { name: [70, 76, 71, 95, 77, 73, 83, 83, 73, 78, 71, 95, 77, 65, 83, 84, 69, 82, 68, 65, 84, 65], data_type: Bit, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 34: BufferDescription { nullable: true, kind: Bit }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 35: ColumnDescription { name: [84, 88, 84, 95, 70, 76, 69, 88, 73, 66, 76, 69, 95, 49], data_type: WVarchar { length: 255 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 35: BufferDescription { nullable: true, kind: Text { max_str_len: 1020 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 36: ColumnDescription { name: [84, 88, 84, 95, 70, 76, 69, 88, 73, 66, 76, 69, 95, 50], data_type: WVarchar { length: 255 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 36: BufferDescription { nullable: true, kind: Text { max_str_len: 1020 } }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC column description for column 37: ColumnDescription { name: [84, 88, 84, 95, 70, 76, 69, 88, 73, 66, 76, 69, 95, 51], data_type: WVarchar { length: 255 }, nullability: Nullable }
2021-02-11T08:13:35+00:00 - DEBUG - ODBC buffer description for column 37: BufferDescription { nullable: true, kind: Text { max_str_len: 1020 } }
2021-02-11T08:13:38+00:00 - INFO - Fetched batch 1 with 100000 rows.
2021-02-11T08:13:41+00:00 - INFO - Fetched batch 2 with 100000 rows.
2021-02-11T08:13:44+00:00 - INFO - Fetched batch 3 with 100000 rows.
2021-02-11T08:13:47+00:00 - INFO - Fetched batch 4 with 100000 rows.
2021-02-11T08:13:50+00:00 - INFO - Fetched batch 5 with 100000 rows.
2021-02-11T08:13:53+00:00 - INFO - Fetched batch 6 with 100000 rows.
2021-02-11T08:13:55+00:00 - INFO - Fetched batch 7 with 100000 rows.
2021-02-11T08:13:58+00:00 - INFO - Fetched batch 8 with 100000 rows.
2021-02-11T08:14:01+00:00 - INFO - Fetched batch 9 with 100000 rows.
2021-02-11T08:14:04+00:00 - INFO - Fetched batch 10 with 100000 rows.
2021-02-11T08:14:06+00:00 - INFO - Fetched batch 11 with 100000 rows.
2021-02-11T08:14:09+00:00 - INFO - Fetched batch 12 with 100000 rows.
thread 'main' panicked at 'ODBC driver indicated must indicate string length correctly.: FromBytesWithNulError { kind: InteriorNul(38) }', C:\Users\runneradmin\.cargo\registry\src\github.com-1ecc6299db9ec823\odbc-api-0.14.0\src\buffers\text_column.rs:56:22
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

2. It's `Microsoft Windows Server 2019 Datacenter` running on the AWS
```bash
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)

[System.Text.Encoding]::Default


Preamble          :
BodyName          : utf-8
EncodingName      : Unicode (UTF-8)
HeaderName        : utf-8
WebName           : utf-8
WindowsCodePage   : 1200
IsBrowserDisplay  : True
IsBrowserSave     : True
IsMailNewsDisplay : True
IsMailNewsSave    : True
IsSingleByte      : False
EncoderFallback   : System.Text.EncoderReplacementFallback
DecoderFallback   : System.Text.DecoderReplacementFallback
IsReadOnly        : True
CodePage          : 65001
```

3. It always fails somewhere between 1 and 2 mln records so it looks like it's a specific value
4. Is the information above enough to diagnose what is wrong? If no, later Today I can try to figure out which column caused that and maybe even get the exact row, but this will probably take some time.

**Comment by pacman82** [comment]
This is already extremely helpful and rules out my initial hypothesis. I can not spend any time on this today, but I can release a version printing the offending value over the weekend (maybe earlier). As you might imagine knowing the value would be very helpful.

**Comment by AdrianSkierniewski** [comment]
@pacman82 I found the exact row and column :) it's ```VersaPlate FFC 4'x8' - 12 Plate Cabine``` -> 'x8'

**Comment by pacman82** [comment]
@AdrianSkierniewski Hi, thanks for researching that. It seems the issue is exactly what the error message said it was. There is an interier `nul` within the string.

Personally I am suprised by the fact that databases allow this, but MSSQL actually seem to be fine with this. I can now easily reproduce the error with an MSSQL database ([see here](https://github.com/pacman82/odbc-api/blob/6eb9d3feaccf9a708d05516c107709f5613012d1/odbc-api/tests/mssql.rs#L806)). There is nothing that forbids databases to store interior nuls, but it means you can store strings what seem shorter to some programs and longer to others. This is the kind of thing that makes me think I should earn my money with Bug bounties.

I do not fully get what you mean by `-> 'x8'` though. Yet the string `VersaPlate FFC 4'x8' - 12 Plate Cabine` makes much sense to me. Mostly because it has 37 letters, placing a `nul` character at the 38 position at the end, as indicated by the original error message. However the driver tells us that the string is in truth longer than that. This raises the question what values are after the terminating zero.

Now that I know this is a possibility I can easily fix the tool. Currently my planned fix is as such: Crazy string in -> crazy string out. Meaning I'll just write the weird string containing interior nuls into the parquet file as is.

That would end this for me, but you probably do not want values like that in your database. Either this is malicious or the code inserting them is buggy.

We can however easily query the extra string at the end casting the value into a `VARBINARY` during the `SELECT`.

Like this: `SELECT CAST(COLUMN_NAME AS VARBINARY) FROM TABLE_NAME;`

The most benign thing we could see is padded zeros at the end. This would just mean that something in the insertion code mistook the value length for a maximum buffer length. If we see anything else, we google an ASCII table and go from there.

**Comment by AdrianSkierniewski** [comment]
@pacman82 Thank you for such a detailed explanation of the problem. I'm surprised as well that MSSQL allows this, but since we're living in an imperfect world I could easily assume that all this weird stuff can and it will eventually happen. I agree that the approach `crazy string in -> crazy string out` is the correct way of solving this problem, so I can take care of this problem on my end.

**Comment by pacman82** [comment]
@AdrianSkierniewski I had some free time on my hands today. Released Version 0.5.7 which does not panic then encountering interior `nul`s.

Pleasure working with you
